cmd/weaver/load_uprobe.go

```diff
@@ -129,6 +147,13 @@ func loadUprobeAndBPFModule(traceContext *functionTraceContext, runtimeContext c
 	output := output{FunctionName: traceContext.FunctionName}
 	var argOutput = make([]outputArg, numberOfArgs)
 	go func() {
+		// First sent values are process info
+		value := <-channel
+		output.ProcInfo = procInfo{}
+		err := output.ProcInfo.unmarshalBinary(value)
+		if err != nil {
+			fmt.Println("failed to unmarshall process info")
+		}
```

Comment:
I seem to be getting [random panics](https://i.imgur.com/q5URWyf.png) when running on the tester binary.

What I believe I overlooked here is that your code assumes only one process will ever trigger this uprobe. The eBPF program (you can actually see it using the `--debug` flag btw) is correctly sending PID/PPID/COMM info every time it triggers. However, that proc info is only read in once, before this loop. So in the loop it's reading the process info into function argument values, and not treating it as new process info. 

I was confused by the fact that I saw different PPIDs, but that's because the uprobes triggered before other ones were installed (I need to get #33 finished). 

I think the easy fix is just to move this logic into the loop. We may want to explore having only one read off the channel since reads off perf buffers [can get lossy](https://stackoverflow.com/questions/60164916/ebpf-packet-monitor-losing-some-packets-when-ping-fed). I think that'd just be creating a struct in eBPF and a corresponding one in Go to marshal it to. That struct would have both proc info, and all the argument info. Might just be too big of a data type though.

Please let me know if that makes sense or not! I can certainly help out with that, just probably not until tomorrow, about to commute home for the day. I really  appreciate your contribution and working with you!

Reply:
Indeed, I did not try to call the test program several times... Included the logic into the loop, does not seems to panic anymore